python/plugins/processing/tests/README.md

```diff
@@ -0,0 +1,96 @@
+Algorithm tests
+===============
+
+To test algorithms you can add entries into `testdata/algorithm_tests.yaml`.
+
+This file is structured with [yaml syntax](http://www.yaml.org/start.html).
+
+A basic test appears under the toplevel key `tests` and looks like this:
+
+  - name: centroid
+    algorithm: qgis:polygoncentroids
+    params:
+      - type: vector
+        location: qgs
+        name: polys.shp
+    results:
+      - id: OUTPUT_LAYER
+        type: vector
+        location: proc
+        name: polys_centroid.geojson
+
+How To
+------
+
+To add a new test you can follow these steps:
```

Comment:
It's in the history, but it seems it is disabled (probably because it was not used, to avoid confusion, but the code is there anyway and you can bring it back just removing one line)

https://github.com/qgis/QGIS/blob/master/python/plugins/processing/gui/HistoryDialog.py#L131